lib/HttpClient.js

```diff
@@ -667,28 +674,54 @@ HttpClient.prototype.request = function request(opts, cb) {
     opts.audit = this.audit;
 
     if (opts.retry === false) {
-        rawRequest(opts, cb);
+        rawRequest(opts, {}, cb);
         return;
     }
 
     var call;
     var retry = cloneRetryOptions(opts.retry);
-
+    var retryInfo = {
+        retries: [],
+        // attempts start at 1, and will increment on per retry
+        attempts: 1
+    };
+    // set keep alive only when retries are enabled
     opts._keep_alive = this._keep_alive;
-    call = backoff.call(rawRequest, opts, cb);
-    call.setStrategy(new backoff.ExponentialStrategy({
-        initialDelay: retry.minTimeout,
-        maxDelay: retry.maxTimeout
-    }));
-    call.failAfter(retry.retries);
-    call.on('backoff', this.emit.bind(this, 'attempt'));
-
-    call.start();
+
+    if (opts.exponentialBackoff === false) {
+        // if no backoff, we have to keep track of num retries ourselves instead
+        // of relying on backoff module.
+        rawRequest(opts, retryInfo, function retryRawRequest(err, req) {
+            if (err && retryInfo.attempts <= retry.retries) {
```

Comment:
Don't you want to check for `statusCode`? We don't want to retry for client error like 404, right?

Reply:
This is already handled in `rawRequest`, but the behavior is that an err is only returned if it's a networking issue. Turning a statusCode into an error actually happens after this.